lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js

```diff
@@ -669,6 +669,70 @@ function factory( dtype ) { // eslint-disable-line max-lines-per-function, stdli
 		return this._buffer[ GETTER ]( idx*BYTES_PER_ELEMENT, this._isLE );
 	});
 
+	/**
+	* Returns a new typed array containing the elements in sorted order.
+	*
+	* @private
+	* @name toSorted
+	* @memberof TypedArray.prototype
+	* @type {Function}
+	* @param {Function} compareFcn - comparison function
+	* @throws {TypeError} `this` must be a typed array
+	* @throws {TypeError} first argument must be a function
+	* @returns {TypedArray} sorted array
+	*/
+	setReadOnly( TypedArray.prototype, 'toSorted', function sort( compareFcn ) {
+		var minIdx;
+		var minVal;
+		var obuf;
+		var buf;
+		var len;
+		var out;
+		var val;
+		var i;
+		var j;
+
+		if ( !isTypedArray( this ) ) {
+			throw new TypeError( format( 'invalid invocation. `this` is not %s %s.', CHAR2ARTICLE[ dtype[0] ], CTOR_NAME ) );
+		}
+		if ( arguments.length === 0 ) {
+			compareFcn = function defaultCompareFcn( a, b ) {
+				if ( a > b ) {
+					return 1;
+				}
+				if ( a < b ) {
+					return -1;
+				}
+				return 0;
+			};
+		} else if ( !isFunction( compareFcn ) ) {
+			throw new TypeError( format( 'invalid argument. First argument must be a function. Value: `%s`.', compareFcn ) );
+		}
+		len = this._length;
+		out = new this.constructor( flag2byteOrder( this._isLE ), len );
+		buf = this._buffer;
+		obuf = out._buffer; // eslint-disable-line no-underscore-dangle
+		for ( i = 0; i < len; i++ ) {
+			obuf[ SETTER ]( i * BYTES_PER_ELEMENT, buf[ GETTER ]( i * BYTES_PER_ELEMENT, this._isLE ), this._isLE );
+		}
+		for ( i = 0; i < len - 1; i++ ) {
+			minIdx = i;
+			minVal = obuf[ GETTER ]( i * BYTES_PER_ELEMENT, this._isLE );
+			for ( j = i+1; j < len; j++ ) {
+				val = obuf[ GETTER ]( j * BYTES_PER_ELEMENT, this._isLE );
+				if ( compareFcn( minVal, val ) > 0 ) {
```

Comment:
Rather than rolling your own sort function, copy to a temporary generic array, sort using the built-in `#.sort` method, and then copy to an output array.